internal/planner/configuration.go

```diff
@@ -181,3 +199,18 @@ func hasShare(cfg smbcc.ConfigSection, k smbcc.Key) bool {
 	}
 	return false
 }
+
+func removeShare(cfg *smbcc.ConfigSection, k smbcc.Key) bool {
+	idx := -1
+	for i := range cfg.Shares {
+		if cfg.Shares[i] == k {
+			idx = i
+			break
+		}
+	}
+	if idx == -1 {
+		return false
+	}
+	cfg.Shares = append(cfg.Shares[:idx], cfg.Shares[idx+1:]...)
+	return true
```

Comment:
I believe the Go community prefers the following code style:

```
	for i := range cfg.Shares {
		if cfg.Shares[i] == k {
			cfg.Shares = append(cfg.Shares[:i], cfg.Shares[i+1:]...)
			return true
		}
	}
	return false
```

Reply:
I had it written that way originally and ran into a weird issue, but it may have been something else I had mistakenly done.  Since I have unit tests for this now  I can try changing it back and seeing what happens.

Reply:
ACK